onDisable ClassNofFound Error

### Expected behavior

No error

### Observed/Actual behavior

When shutting down the plugin, an error occurs.

```kotlin
[08:17:53 ERROR]: Error occurred while disabling surf-chat-bukkit v1.21.4-1.0.0-SNAPSHOT
java.lang.NoClassDefFoundError: dev/slne/surf/chat/bukkit/SurfChatBukkit$onDisable$1
        at surf-chat-bukkit-1.21.4-1.0.0-SNAPSHOT-all.jar/dev.slne.surf.chat.bukkit.SurfChatBukkit.onDisable(SurfChatBukkit.kt:95) ~[surf-chat-bukkit-1.21.4-1.0.0-SNAPSHOT-all.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:286) ~[paper-api-1.21.4-R0.1-SNAPSHOT.jar:?]
        at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.disablePlugin(PaperPluginInstanceManager.java:237) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.disablePlugins(PaperPluginInstanceManager.java:161) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.disablePlugins(PaperPluginManagerImpl.java:97) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at org.bukkit.plugin.SimplePluginManager.disablePlugins(SimplePluginManager.java:541) ~[paper-api-1.21.4-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.CraftServer.disablePlugins(CraftServer.java:626) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at net.minecraft.server.MinecraftServer.stopServer(MinecraftServer.java:1007) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at net.minecraft.server.dedicated.DedicatedServer.stopServer(DedicatedServer.java:757) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1294) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:310) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.ClassNotFoundException: dev.slne.surf.chat.bukkit.SurfChatBukkit$onDisable$1
        at io.papermc.paper.plugin.entrypoint.classloader.PaperPluginClassLoader.loadClass(PaperPluginClassLoader.java:146) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at io.papermc.paper.plugin.entrypoint.classloader.PaperPluginClassLoader.loadClass(PaperPluginClassLoader.java:107) ~[paper-1.21.4.jar:1.21.4-222-9b1798d]
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
        ... 12 more
```

### Steps/models to reproduce

- Start a server with surf-chat (feat/add-cross-server-messages)
- stop the server

### Other

_No response_

--- a/surf-chat-bukkit/src/main/kotlin/dev/slne/surf/chat/bukkit/SurfChatBukkit.kt
+++ b/surf-chat-bukkit/src/main/kotlin/dev/slne/surf/chat/bukkit/SurfChatBukkit.kt
@@ -85,19 +85,17 @@ class SurfChatBukkit(): SuspendingJavaPlugin() {
         LuckPermsExtension.loadApi()
     }
 
-    override fun onDisable() {
-        // Note: measureTimeMillis may throw java.lang.NoClassDefFoundError: dev/slne/surf/chat/bukkit/SurfChatBukkit$onDisable$ms$1$1.
-        // This issue requires further investigation to identify the root cause and implement a fix.
-        val start = System.currentTimeMillis()
+    override suspend fun onDisableAsync() {
+        val ms = measureTimeMillis {
+            chatMotdService.saveMotd()
+            filterService.saveMessageLimit()
 
-        chatMotdService.saveMotd()
-        filterService.saveMessageLimit()
-
-        runBlocking {
-            databaseService.saveAll()
+            runBlocking {
+                databaseService.saveAll()
+            }
         }
 
-        logger.info { "Successfully disabled in ${System.currentTimeMillis() - start}ms!" }
+        logger.info { "Successfully disabled in ${ms}ms!" }
     }
 
     fun getTeamMembers(): ObjectSet<Player> = serverPlayers.filter { it.hasPermission("surf.chat.command.teamchat") }.toObjectSet()

